src/utils/error/player-error.js

```diff
@@ -0,0 +1,83 @@
+//@flow
+import getLogger, {getLogLevel, LogLevel} from '../logger'
+import {Severity} from './severity'
+import type {SeverityType} from './severity'
+import {Code} from './code'
+import type {CodeType} from './code'
+import {Category} from './category'
+import type {CategoryType} from './category'
+
+
+const CLASS_NAME: string = 'Error';
+
+const UNKNOWN: string = 'UNKNOWN'
+
+/**
+ * @classdesc This is a description of the error class.
+ */
+export default class Error {
+  severity: number;
+  category: number;
+  code: number;
+  data: any;
+  /**
+   * @enum {number}
+   */
+  static Severity: SeverityType = Severity;
+  /**
+   * @enum {number}
+   */
+  static Category: CategoryType = Category;
+  /**
+   * @enum {number}
+   */
+  static Code: CodeType = Code;
+  static _logger: any = getLogger(CLASS_NAME);
+
+  /**
+   * @constructor
+   * @param {number} severity - error's severity
+   * @param {number} category - error's category.
+   * @param {number} code - error's code.
+   * @param {any} data - additional data for the error.
+   */
+  constructor(severity: number, category: number, code: number, data: any = {}) {
+    this.severity = severity;
+    this.category = category;
+    this.code = code;
+    this.data = data;
+    if (getLogLevel(CLASS_NAME) === LogLevel.DEBUG) {
+      this._createReadableError(category, code, data);
+    }
+  }
+
+  /**
+   * creates a readable form of the error
+   * @param {number} category - error's category.
+   * @param {number} code - error's code.
+   * @param {any} data - additional data for the error.
+   * @returns {void}
+   */
+  _createReadableError(category: number, code: number, data: any = {}): void {
+    let codeName = this._getKey(Error.Code, code);
+    let categoryName = this._getKey(Error.Category, category);
+    Error._logger.debug('Player error ' + categoryName + '.' + codeName + ' (' + JSON.stringify(data) + ')');
```

Comment:
don't stringify the data like this, the can cause long and unreadable log.
Attach the data as another param to the log message which will the browser dev tool can display as an object.
`Error._logger.debug('Player error ' + categoryName + '.' + codeName' ,  data);`